tests/cross/src/main/scala/tests/BasePCSuite.scala

```diff
@@ -1,105 +1,153 @@
+      .create()
+      .withClassifiers(Set("sources").asJava)
+      .withDependencies(
+        Dependency.of(
+          "org.scala-lang",
+          "scala-library",
           // NOTE(gabro): we should ideally just use BuildoInfoVersions.scalaVersion
           // but using the 2.11 stdlib would cause a lot tests to break for little benefit.
           // We can remove this switch once we drop support for 2.11
-          BuildInfoVersions.scalaVersion match {
+          scalaVersion match {
             case v if v.startsWith("2.13") => v
             case v if v.startsWith("2.12") => v
             case _ => BuildInfoVersions.scala212
           }
         )
       )
-      .run()
+      .fetch()
+      .asScala
     sources.foreach { jar => index.addSourceJar(AbsolutePath(jar)) }
   }
 
+  protected lazy val presentationCompiler: PresentationCompiler = {
+    val binaryVersion = ScalaDependencies.createBinaryVersion(scalaVersion)
+    val scalaDependencies =
+      if (binaryVersion.startsWith("0."))
+        ScalaDependencies.scala3(scalaVersion)
+      else
+        ScalaDependencies.scala2(scalaVersion)
+
+    val fetch = Fetch
+      .create()
+
+    val allDependencies = scalaDependencies ++ extraDependencies(scalaVersion)
+
+    allDependencies.foreach(fetch.addDependencies(_))
+    val libraries: Seq[Path] = fetch
+      .fetch()
+      .asScala
+      .map(_.toPath())
+
+    val myclasspath: Seq[Path] = libraries.filterNot { path =>
+      val filename = path.toString()
+      filename.contains("scala-reflect") ||
+      filename.contains("scala-compiler") ||
+      filename.contains("scala-xml")
+    }
+
+    if (requiresJdkSources)
+      JdkSources().foreach(jdk => index.addSourceJar(jdk))
+    if (requiresScalaLibrarySources)
+      indexScalaLibrary(index, scalaVersion)
+    val indexer = new Docstrings(index)
+    val search = new TestingSymbolSearch(
+      ClasspathSearch.fromClasspath(myclasspath),
+      new Docstrings(index),
+      workspace,
+      index
+    )
+
+    val scalacOpts = scalacOptions(myclasspath)
+
+    new ScalaPresentationCompiler()
+      .withSearch(search)
+      .withConfiguration(config)
+      .withExecutorService(executorService)
+      .withScheduledExecutorService(executorService)
+      .newInstance("", myclasspath.asJava, scalacOpts.asJava)
+  }
+
+  val tmp: AbsolutePath = AbsolutePath(Files.createTempDirectory("metals"))
+
+  override def afterAll(): Unit = {
+    presentationCompiler.shutdown()
+    RecursivelyDelete(tmp)
+    executorService.shutdown()
+  }
+
+  def testPc(name: TestOptions, ignoredScalaVersions: Set[String] = Set.empty)(
+      fn: PresentationCompiler => Unit
+  ): Unit = {
+    if (excludedScalaVersions(scalaVersion)) {
+      test(name.ignore) {}
+    } else {
+      val testOptions: TestOptions =
+        name.withName(s"${name.name}_$scalaVersion")
+      val possiblyIgnored =
+        if (ignoredScalaVersions(scalaVersion)) testOptions.ignore
```

Comment:
This can be replaced with a custom tag + test transform
```scala
case class IgnoreScalaVersion(versions: String*) extends Tag("NoScalaVersion")
  override def munitTestTransforms = super.munitTestTransforms ++ List(
    new TestTransform("IgnoreScalaVersion", { test =>
      val isIgnoredScalaVersion = test.tags
         .collect { case IgnoreScalaVersion(versions) => versions.toSet }
         .getOrElse(Set.empty)
      if (isIgnoredScalaVersion(scalaVersion)) test.ignore
      else test
    })
  )

test("foo".tag(IgnoreScalaVersion("2.11.12")) { ... }
```

Reply:
Added it here: https://github.com/scalameta/metals/pull/1523/files#diff-713c81e0230fa147e415ec5325807f59R133-R146

Had to change the last part to:
```
https://github.com/scalameta/metals/pull/1523/files#diff-713c81e0230fa147e415ec5325807f59R133-R146
```